include/battle.h

```diff
@@ -790,6 +789,92 @@ struct AiBattleData
     u8 choiceWatch:1;
     u8 padding:6;
 };
+union PACKED StatAnimArg
+{
+    u8 value;
+    u16 value_u16; // So that we can cast from a u16
+    struct PACKED {
+        u8 isNegative:1;
+        u8 harshly:1;
+        u8 stat:6;
+    };
+};
+
+union TRANSPARENT StatFlags
+{
+    int raw;
+    u8 allStats;
+    u32 value;
+    struct {
+        u32 unused:1; // HP
+        u32 attack:1;
+        u32 defense:1;
+        u32 speed:1;
+        u32 spAttack:1;
+        u32 spDefense:1;
+        u32 accuracy:1;
+        u32 evasion:1;
+        u32 padding:24;
+    };
+};
+
+union PACKED StatChangerKey
+{
+    u8 value;
+    struct PACKED {
+        u8 skipFailStrings:1; // Skips "couldn't go any lower/higher!" strings
+        u8 attack:1;
+        u8 defense:1;
+        u8 speed:1;
+        u8 spAttack:1;
+        u8 spDefense:1;
+        u8 accuracy:1;
+        u8 evasion:1;
+    };
+    struct PACKED {
+        u8 backupSaveBit:1; // Marks a savedStatChanger even if allStats = 0
+        u8 allStats:7;
+    };
+};
+
+struct MoveEffectResult
+{
+    // Should be populated at initialisation
+    const u8 *pushInstr; // Instruction that will be pushed if battlescriptPush is set
+    const u16 move; // gCurrentMove
+    const u16 battlerItem; // gEffectBattler held item
```

Comment:
```suggestion
    const u16 heldItem; // gEffectBattler held item
```

We generally follow the naming convention of `heldItem` for battlers. If we explicitly want to say heldItem of attacker or target we add the suffix Atk/Def, so `heldItemAtk / heldItemDef`

I know it isn't consistent but